ckanext/validation/jobs.py

```diff
@@ -59,37 +66,38 @@ def run_validation_job(resource):
         {'ignore_auth': True}, {'id': resource['package_id']})
 
     source = None
-    if resource.get('url_type') == 'upload':
+    if resource.get(u'url_type') == u'upload':
         upload = uploader.get_resource_uploader(resource)
         if isinstance(upload, uploader.ResourceUpload):
```

Comment:
This should check if the upload has a `get_path` attribute instead, so that other extended ResourceUploaders (like the CloudStorage plugin) can work.

Reply:
`if isinstance(upload, uploader.ResourceUpload):` -->> `if hasattr(upload, 'get_path'):`

Reply:
will come in the next cut 

also will do upload check from qld-gov-au, 
```
def is_uploaded_file(upload):
    return isinstance(upload,
                      uploader.ALLOWED_UPLOAD_TYPES) and upload.filename
```